builddefs/common_features.mk

```diff
@@ -589,6 +589,15 @@ ifneq ($(strip $(CUSTOM_MATRIX)), yes)
     endif
 endif
 
+VALID_CUSTOM_ENCODER_TYPES:= yes no
```

Comment:
I think that this would be better as `ENCODER_DRIVER`, as that fits more with the general design that exists. 

Default to `gpio`, and have a `custom` option.

This way, it would better leverage the existing code, without cludgy hacks like `SRC += encoder.c`.

Especially as this prevents encoder map from working properly.

Reply:
A `SRC += encoder.c` would still be needed with an `ENCODER_DRIVER = custom`, would it not? Otherwise the file containing the custom implementation would not be compiled.

I have not tested it, but as I understand it the encoder map _should_ still work. Could you clarify why it would break?

Reply:
No need for custom `encoder.c` so long as any of the existing source files provide an implementation of the functions -- in most cases that would likely be the actual `keyboard.c` file.

Encoder map doesn't look like it'd be affected by the change, but I'd want it tested nonetheless.

`ENCODER_DRIVER` is the way forward, though. All the abstractions are headed that way.

Reply:
Basically, it would be something like matrix lite, or pointing device driver, etc.  

Just needs to provide a way to set the init and reading data, and let the core code (encoder.c) just worry about handling.

And definitely interested in this, as it would make the ploopy scroll code a lot simpler.